### ISSUE TITLE
Assertion in InputInt(label) after calling GetID(label) after value_changed

### ISSUE THREAD

**Issue by instinkt900** [ocornut/imgui#5181]
**Version/Branch of Dear ImGui: 1.88 WIP**

Version: 1.88 WIP
Branch: Docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl.cpp + imgui_impl_sdl_renderer.cpp
Compiler: MSVC 1929
Operating System: Win64

**My Issue/Question:**

I'm using ImGuiIDs to keep track of some stateful data and noticed that when I use GetID() before using InputInt() if I try to change the value using keyboard input I get an assertion in ImGui::MarkItemEdited() on `IM_ASSERT(g.ActiveId == id || g.ActiveId == 0 || g.DragDropActive);` where g.ActiveId seems to be the correct ID for my element but id is 0.
A minimal example of a repo is below.
```
static int testInt = 60;
auto const id = ImGui::GetID("Test Int");
ImGui::InputInt("Test Int", &testInt);
```
When selecting the input and hitting a key the assertion is fired. Note that it doesn't happen with InputFloat or InputInt with 0 as the stepping value (stepping buttons removed).

**Comment by ocornut** [comment]
Wow, that's a great bug.
`EndGroup()` compares the value of `g.ActiveIdIsAlive` before and after in order to tell if the group contained the active id, but `GetID()` by default sets that flag, which messes with this behavior.

A possible direction for a fix would be to move the call to `KeepAliveID()` from the 3 `GetID()` and `GetIDFromRectangle()` to `ItemAdd()` but I seriously worry at how many side-effects this would lead to (our regressions tests are passing to that's not TOO bad but pretty sure there will be a an issue somewhere).

If you feel like running with this change it would be useful..

**Comment by instinkt900** [comment]
Added the suggested change to my own project. It fixes the bug with no obvious side effects but I only currently have little coverage of features.

**Comment by ocornut** [comment]
Well it breaks a few things like manual resizing of windows.. 
I am currently reviewing every call to `GetID()` and will fix things accordingly.

**Comment by ocornut** [comment]
Pushed a fix 90386780 + subsequent cleanup 6d15a50
Thanks for reporting!

**Comment by instinkt900** [comment]
I cherry picked this fix into the docking branch and noticed now if I have a window docked with the tab bar hidden, I can no longer click the little corner triangle to bring back the tab bar. It's the only issue I've noticed so far.
